core/src/main/java/org/hibernate/ogm/dialect/spi/DialectNamingAdaptor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.spi;
+
+/**
+ * The Dialect can validate or adjust the names which will be used to actually persist into the database "tables".
```

Comment:
I would put it a bit more generically: "... the names of structures in the datastores", leaving the door open for future amendment. Also can you add a note that one should not implement this interface directly but rather extend the base class?